File: app/src/main/java/ru/eugen/noteseugen/data/CardsSource.java
package ru.eugen.noteseugen.data;

public interface CardsSource {
    CardsSource init(CardsSourceResponse cardsSourceResponse);
    Card getCard(int position);
    int size();
    void deleteCard(int position);
    void updateCard(int position, Card card);
    void addCard(Card card);
    void clearCard();
}

File: app/src/main/java/ru/eugen/noteseugen/data/CardsSource.kt
package ru.eugen.noteseugen.data;

interface CardsSource {
    fun init(cardsSourceResponse:CardsSourceResponse):CardsSource
    fun getCard(position:Int):Card
    fun size():Int
    fun deleteCard(position:Int):Unit
    fun updateCard(position:Int, card:Card):Unit
    fun addCard(card:Card):Unit
    fun clearCard():Unit
}
